File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const numbers = {}
        for (const number of nums) {
            numbers[number] = (numbers[number] || 0) + 1
        }
        // const numbers = _.countBy() // Easy to count number
        const list = Object.entries(numbers).map(([num, count]) => [count, num]).sort((a, b) => b[0] -a[0])
        // return list.slice(0, k)
        return Array.from({ length: Math.min(k, list.length) }, (_, i) => list[i][1]);
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) return 0;
        Set<Integer> uniqueSet = removeDuplicates(nums);
        int longestConsecutiveNumber = 1;
        List<Integer> numbers = new ArrayList<>(uniqueSet);
        Collections.sort(numbers);
        int i = 0;
        int count = 1;
        while(i < numbers.size() - 1) {
            if((numbers.get(i) + 1) == numbers.get(i + 1)) {
                count += 1; 
            } else {
                longestConsecutiveNumber = Math.max(count, longestConsecutiveNumber);
                count = 1;
            }
            i++;
        }
        return Math.max(count, longestConsecutiveNumber);
    }

    public static Set<Integer> removeDuplicates(int[] nums) {
        Set<Integer> uniqueNumbers = new HashSet<>();
        for (int num : nums) {
            uniqueNumbers.add(num); // Automatically ignores duplicates
        }
        return uniqueNumbers;
    }
}
